paddle/parameter/Argument.cpp

```diff
@@ -572,4 +572,26 @@ void Argument::subArgFrom(const Argument& input, size_t offset, size_t height,
   }
 }
 
+void Argument::idsToSparseMatrix(int width, bool useGpu) {
+  if (ids) {
+    CHECK(!value);
+    int height = ids->getSize();
+    int nnz = height;
+    auto rows = IVector::create(height + 1, useGpu);
+    auto cols = IVector::create(nnz, useGpu);
+    rows->setElement(0, 0);
+    for (int i = 0; i < height; i ++) {
+      int id = ids->getElement(i);
+      CHECK_LT(id, width);
+      rows->setElement(i + 1, i + 1);
+      cols->setElement(i, id);
+    }
+    value = Matrix::createSparseMatrix(
```

Comment:
This constructor will not copy value, rows and cols. It only keep of raw pointer to these buffers. If the original IVector is destructed, those pointer will become invalid.
Use setRow() or copyFrom to create the sparse matrix.